src/asmcnc/production/lowerbeam_calibration_jig/lowerbeam_calibration_4.py

```diff
@@ -65,7 +70,22 @@ def __init__(self, **kwargs):
     def enter_prev_screen(self):
         self.sm.current = 'lbc2'
 
+    def validate_serial_number(self, serial):
+        expression = '(xl)\d{4}'
+        pattern = re.compile(expression)
+        match = bool(pattern.match(serial))
+
+        return match
+
     def enter_next_screen(self):
+        serial_number = self.serial_no_input.text.replace(' ', '').lower()
+
+        validated = self.validate_serial_number(serial_number)
+
+        if not validated:
+            self.error_label.text = 'Serial number invalid'
+            return
+
         try:
             self.send_calibration_payload(TMC_Y1)
             self.send_calibration_payload(TMC_Y2)
```

Comment:
should the set_serial_no() functions be set_serial_no(serial_number) instead of set_serial_no(self.serial_no_input.text) ?